app/Repositories/CacheUserRepository.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Gladiator\Repositories;
+
+use Gladiator\Models\User;
+use Illuminate\Support\Facades\Cache;
+
+class CacheUserRepository implements UserRepositoryInterface
+{
+    /**
+     * UserRepositoryInterface instance.
+     *
+     * @var \Gladiator\Repositories\UserRepositoryInterface
+     */
+    protected $database;
+
+    /**
+     * Create new CacheUserRepository instance.
+     *
+     * @param UserRepositoryInterface $database
+     */
+    public function __construct(UserRepositoryInterface $database)
+    {
+        $this->database = $database;
+    }
+
+    /**
+     * Create and cache a new user.
+     *
+     * @param  object  $account
+     * @return object
+     */
+    public function create($account)
+    {
+        $user = $this->database->create($account);
+
+        $this->resolveUpdatedRoles($user->id, $user->role);
+
+        $user = $this->find($user->id);
+
+        return $user;
+    }
+
+    /**
+     * Find the specified resource in cache or default to database lookup.
+     *
+     * @param  string  $id  Northstar ID
+     * @return object
+     */
+    public function find($id)
+    {
+        $user = Cache::get($id);
+
+        if (! $user) {
+            $user = $this->database->find($id);
+
+            $this->store($user->id, $user);
+        }
+
+        return $user;
+    }
+
+    /**
+     * Get collection of users from cache by specified role or default to database lookup.
+     *
+     * @param  string $role
+     * @return \Illuminate\Support\Collection
+     */
+    public function getAllByRole($role)
+    {
+        $key = $role ? $role : 'contestant';
+
+        $ids = $this->retrieve($key . ':ids');
+
+        if ($ids) {
+            $users = $this->retrieveMany($ids);
+            $users = $this->resolveMissingUsers($users);
+            $users = collect(array_values($users));
+
+            return $users;
+        }
+
+        $users = $this->database->getAllByRole($role);
+
+        if ($users->count()) {
+            $ids = $users->pluck('id')->toArray();
+            $collection = collect($users)->keyBy('id')->toArray();
+
+            $this->store($key . ':ids', $ids);
+            $this->storeMany($collection);
+        }
+
+        return $users;
+    }
+
+    /**
+     * Update the specified user's data and resolve the cache.
+     *
+     * @param  \Gladiator\Http\Requests\UserRequest $request
+     * @param  string $id  Northstar ID
+     * @return object
+     */
+    public function update($request, $id)
+    {
+        $user = $this->database->update($request, $id);
+
+        $this->resolveUpdatedRoles($id, $request->role);
+
+        return $user;
+    }
+
+    /**
+     * Remove an item from the cache.
+     *
+     * @param  string  $key
+     * @return void
+     * @TODO: Might be best to return a bool like the Laravel class does?
+     */
+    protected function forget($key)
```

Comment:
These feel a little bit like premature optimization, since they're just one-line aliases to framework methods... I could be wrong, but let's just keep an eye on whether or not they seem useful in the future.